sklearn/linear_model/_sgd_fast.pyx.tp

```diff
@@ -359,37 +396,116 @@ cdef class SquaredEpsilonInsensitive(Regression):
     def __reduce__(self):
         return SquaredEpsilonInsensitive, (self.epsilon,)
 
+def _plain_sgd(
```

Comment:
This is generated compiled functions that take float32 weights and SequentialDataset64, which are not used together. If this is the case, I prefer to update the call site from Python and direct to `_plain_sgd32` or `_plain_sgd64`. For example:

https://github.com/scikit-learn/scikit-learn/blob/1042757565b85960c42e97ef4a21d9943e7cb48e/sklearn/linear_model/_stochastic_gradient.py#L446

becomes:

```python
_plain_sgd = _plain_sgd32 if coef.dtype == np.float32 else _plain_sgd64
coef, intercept, average_coef, average_intercept, n_iter_ = _plain_sgd(
```

This way, we do not need to define `_plain_sgd` at all in the Cython file.